File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        h = {}
        for i in range(len(nums)):
            h[target - nums[i]] = i
        for i in range(len(nums)):
            if nums[i] in h:
                if i != h[nums[i]]:
                    if i > h[nums[i]]:
                        return [h[nums[i]],i]
                    else:
                        return [i,h[nums[i]]]

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> seen;
        int max_seq = 0;

        for (int num : nums) {
            if (!seen[num]) {
                seen[num] = seen[num - 1] + seen[num + 1] + 1;
                seen[num - seen[num - 1]] = seen[num];
                seen[num + seen[num + 1]] = seen[num];
                max_seq = max(max_seq, seen[num]);
            }
        }
        return max_seq;
    }
};